### ISSUE TITLE
How to write CoordinateMappingProvider

### ISSUE THREAD

**Issue by github-yizhang** [facebookresearch/silk#66]
Hello, this is an excellent work! I tried to use SqueezeNet as the backbone of SILK, But I am confused how to  write CoordinateMappingProvider in the Fire module. This because VGG has only one path, it is easy to write. But the Fire Module has two path parallelly, I don't know how to deal with the parallel mappings. Could you help me about that? Thank you very much! 

    class Fire(torch.nn.Module, CoordinateMappingProvider):
        def __init__(self, in_channels: int,
        out_channels: int,
        kernel_size: int,
        normalization_fn,
        non_linearity: str = "relu",
        padding: int = 1)-> torch.nn.Module:
            super(Fire, self).__init__()

        if non_linearity == "relu":
            non_linearity = torch.nn.ReLU(inplace=True)
        else:
            raise NotImplementedError
        
        squeeze_planes = out_channels // 4
        self.conv1 = torch.nn.Conv2d(in_channels, squeeze_planes, kernel_size=1, stride=1)
        self.bn1 = torch.nn.BatchNorm2d(squeeze_planes)
        self.relu1 = torch.nn.ReLU(inplace=True)
        self.conv2 = torch.nn.Conv2d(squeeze_planes, out_channels // 2, kernel_size=1, stride=1)
        self.bn2 = torch.nn.BatchNorm2d(out_channels // 2)
        self.conv3 = torch.nn.Conv2d(squeeze_planes, out_channels // 2, kernel_size=3, stride=1, padding=1)
        self.bn3 = torch.nn.BatchNorm2d(out_channels // 2)
        self.relu2 = torch.nn.ReLU(inplace=True)

        # using MSR initilization
        for m in self.modules():
            if isinstance(m, torch.nn.Conv2d):
                n = m.kernel_size[0] * m.kernel_size[1] * m.in_channels
                m.weight.data.normal_(0, math.sqrt(2./n))
    
    def mappings(self):
        mapping = Identity()
        ????

        return mapping

    def forward(self, x):
        x = self.conv1(x)
        x = self.bn1(x)
        x = self.relu1(x)
        out1 = self.conv2(x)
        out1 = self.bn2(out1)
        out2 = self.conv3(x)
        out2 = self.bn3(out2)
        out = torch.cat([out1, out2], 1)
        out = self.relu2(out)
        return out

**Comment by AliYoussef97** [comment]
I am not entirly sure if this is correct, the authors could correct me if I am wrong.

From what I understand mapping relates input image coordinates to feature map coordinates.

Since out1 and out2 essentially result in the same output shape (kernel=1 + stride 1 + padding 0 == kernel=3+ stride=1 + padding=1), and seince concattination+relu does not change the shape further, you could poternailly define mappings as:

```
def mappings(self):
    mapping = mapping_from_torch_module(self.conv1)
    mapping = mapping + mapping_from_torch_module(self.bn1)
    mapping = mapping + mapping_from_torch_module(self.relu1)
    mapping = mapping + mapping_from_torch_module(self.conv2)
    mapping = mapping + mapping_from_torch_module(self.bn2)
    mapping = mapping + mapping_from_torch_module(self.relu2) # I do not think this is needed since relu will return identity.
    return mapping
```

Again, I might be completely wrong and authors would be more helpful.

**Comment by github-yizhang** [comment]
Thank you very much! I understand what you have said. All I need to do is to define an end-to-end mapping path. I define the mapping as you said, and the experiemental results seem correct. I really appreciate your help and hope this issue could help people like me. The issue is closing now!
